chromadb/test/conftest.py

```diff
@@ -550,21 +558,33 @@ def sqlite() -> Generator[System, None, None]:
 
 def sqlite_persistent() -> Generator[System, None, None]:
     """Fixture generator for segment-based API using persistent Sqlite"""
-    with tempfile.TemporaryDirectory() as save_path:
-        settings = Settings(
-            chroma_api_impl="chromadb.api.segment.SegmentAPI",
-            chroma_sysdb_impl="chromadb.db.impl.sqlite.SqliteDB",
-            chroma_producer_impl="chromadb.db.impl.sqlite.SqliteDB",
-            chroma_consumer_impl="chromadb.db.impl.sqlite.SqliteDB",
-            chroma_segment_manager_impl="chromadb.segment.impl.manager.local.LocalSegmentManager",
-            allow_reset=True,
-            is_persistent=True,
-            persist_directory=save_path,
-        )
-        system = System(settings)
-        system.start()
-        yield system
-        system.stop()
+    save_path = tempfile.TemporaryDirectory()
+    settings = Settings(
+        chroma_api_impl="chromadb.api.segment.SegmentAPI",
+        chroma_sysdb_impl="chromadb.db.impl.sqlite.SqliteDB",
+        chroma_producer_impl="chromadb.db.impl.sqlite.SqliteDB",
+        chroma_consumer_impl="chromadb.db.impl.sqlite.SqliteDB",
+        chroma_segment_manager_impl="chromadb.segment.impl.manager.local.LocalSegmentManager",
+        allow_reset=True,
+        is_persistent=True,
+        persist_directory=save_path.name,
+    )
+    system = System(settings)
+    system.start()
+    yield system
+    system.stop()
+
+    try:
```

Comment:
Do we need this? My understanding was that when a TemporaryDirectory is GC'ed it is cleaned up?

Reply:
I don't think we'd be able to catch and ignore errors on Windows then